core/request.go

```diff
@@ -169,46 +179,59 @@ func (r *RequestAccessor) RequestFromEvent(ctx context.Context, req events.APIGa
 		log.Println(err)
 		return nil, err
 	}
-
 	for h := range req.Headers {
 		httpRequest.Header.Add(h, req.Headers[h])
 	}
+	return httpRequest, nil
+}
 
-	apiGwContext, err := json.Marshal(req.RequestContext)
+// AddToHeader adds API gateway context and stage variables to headers on http request
+func AddToHeader(req *http.Request, apiGwRequest events.APIGatewayProxyRequest) (*http.Request, error) {
+	stageVars, err := json.Marshal(apiGwRequest.StageVariables)
 	if err != nil {
-		log.Println("Could not Marshal API GW context for custom header")
+		log.Println("Could not marshal stage variables for custom header")
 		return nil, err
 	}
-	stageVars, err := json.Marshal(req.StageVariables)
+	req.Header.Add(APIGwStageVarsHeader, string(stageVars))
+	apiGwContext, err := json.Marshal(apiGwRequest.RequestContext)
 	if err != nil {
-		log.Println("Could not marshal stage variables for custom header")
-		return nil, err
+		log.Println("Could not Marshal API GW context for custom header")
+		return req, err
 	}
-	httpRequest.Header.Add(APIGwContextHeader, string(apiGwContext))
-	httpRequest.Header.Add(APIGwStageVarsHeader, string(stageVars))
+	req.Header.Add(APIGwContextHeader, string(apiGwContext))
+	return req, nil
+}
 
+// AddToContext adds context, API gateway context and stage variables to context on http request
+func AddToContext(ctx context.Context, req *http.Request, apiGwRequest events.APIGatewayProxyRequest) *http.Request {
```

Comment:
This likewise might not need to be exported.

Reply:
Agreed. Fixed.